entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt

```diff
@@ -268,61 +278,61 @@ class EntityService(
      */
     fun getFullEntityById(entityId: String): ExpandedEntity {
         val entity = entityRepository.getEntityCoreById(entityId)
-        val resultEntity = entity.serializeCoreProperties()
+        var resultEntity = entity.serializeCoreProperties()
 
         // TODO test with a property having more than one relationship (https://redmine.eglobalmark.com/issues/848)
         entityRepository.getEntitySpecificProperties(entityId)
-            .groupBy {
-                (it["property"] as Property).id
-            }
-            .values
-            .forEach {
-                val (propertyKey, propertyValues) = buildPropertyFragment(it, entity.contexts)
-                resultEntity[propertyKey] = propertyValues
-            }
+        .groupBy {
+            (it["property"] as Property).id
+        }
+        .values
+        .forEach {

```

Comment:
```
.map {  buildInstanceFragment(it, entity.contexts) }
.groupBy { it._1 }
forEach { resultEntity[it._1] = it._2 }
```
Doesn't this work and avoid to make resultEntity a var ?

Reply:
I think it should be: 
```
.map {
    buildInstanceFragment(it, entity.contexts)
}
.groupBy { it.first }
.forEach {
    it.value.forEach {
    resultEntity[it.first] = it.second
    }
}
```
I tried this but it keeps only the last instance in the resultEntity.